pkg/commonvpc/vpc.go

```diff
@@ -179,15 +189,68 @@ func (s *Subnet) AddressRange() *ipblock.IPBlock {
 	return s.IPblock
 }
 func (s *Subnet) IsPrivate() bool {
-	return s.isPrivate
+	return s.subnetExpose == privateExpose
 }
 func (s *Subnet) SetIsPrivate(isPrivate bool) {
-	s.isPrivate = isPrivate
+	s.subnetExpose = publicExpose
+	if isPrivate {
+		s.subnetExpose = privateExpose
+	}
 }
 func (s *Subnet) SynthesisKind() spec.ResourceType {
 	return spec.ResourceTypeSubnet
 }
 
+// ////////////////////////////////////////////////////////////////////////////////
+// privateSubnetRule is the implementation of PrivateSubnetRule
+// it holds the information on the influence of the subnet on the connectivity.
+// its relevant only for providers that allow private subnets (aws)
+type privateSubnetRule struct {
+	subnet    vpcmodel.Subnet
+	src, dst  vpcmodel.Node
+	isIngress bool
+}
+
+func newPrivateSubnetRule(subnet vpcmodel.Subnet, src, dst vpcmodel.Node, isIngress bool) vpcmodel.PrivateSubnetRule {
+	return &privateSubnetRule{subnet, src, dst, isIngress}
+}
+
+func (psr *privateSubnetRule) Deny(isIngress bool) bool {
+	return isIngress == psr.isIngress && psr.subnet.IsPrivate()
```

Comment:
This is a bit confusing: here you also have to verify that `src/dst` is external
Specifically, for `ingress`, `src` has to be external and for `egress`, `dst` has to be external
I understand you are not doing it here since the rule will be nil if src and dst are both internal.

Please document this. Something in the spirit:
// Note that privateSubnetRule is defined (non nil) only when relevant: public/private subnet supported by the platform and src/dst is external

Reply:
Please add the following comment before this func:
// Note that the func is called only when relevant (platform supporting private subnet and connection to/from internet)

Reply:
fixed